frontend/src/metabase/visualizations/components/Visualization/Visualization.tsx

```diff
@@ -101,24 +216,51 @@ const deriveStateFromProps = props => {
   };
 };
 
-class Visualization extends PureComponent {
-  constructor(props) {
+class Visualization extends PureComponent<
+  VisualizationProps,
+  VisualizationState
+> {
+  private _resetHoverTimer: number | null = null;
+
+  static defaultProps = {
```

Comment:
Not sure how to properly type it with `defaultProps`, tried a few approaches, had issues with both. So inside the component we still have nullable values and the logic that check for nullable